internal/pkg/aws/ec2/ec2.go

```diff
@@ -437,3 +438,33 @@ func (idx *routeTableIndex) IsPublicSubnet(subnetID string) bool {
 	}
 	return idx.mainTable.HasIGW()
 }
+
+// ManagedPrefixListId returns the PrefixListId of a prefix list queried by name.
+func (c *EC2) ManagedPrefixListId(prefixListName string) (*string, error) {
+	prefixListOutput, err := c.client.DescribeManagedPrefixLists(&ec2.DescribeManagedPrefixListsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("prefix-list-name"),
+				Values: aws.StringSlice([]string{prefixListName}),
+			},
+		},
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("query returned error: %w", err)
+	}
+
+	if len(prefixListOutput.PrefixLists) == 0 {
+		return nil, fmt.Errorf("cannot find any prefix list with name: %s", prefixListName)
+	}
+
+	if len(prefixListOutput.PrefixLists) > 1 {
```

Comment:
Out of curiosity - when would this happen and why do we want exactly one `PrefixLists`?

Reply:
We mostly want one `PrefixList` to simplify handling the ID in logic for the stack. We can't return a list of ID's because we have no way to confirm which ID is the one we need (in the event of a query returning multiple). If I were to return all of them, we'd have to turn the list of prefix lists in their entirety to maybe have the information to separate them, so I thought it would be simpler to require querying for a unique ID, and to return an error if the query doesn't result in pulling a unique one.

Now I don't think it would ever happen, at least not in this use case, so it's mostly just a case for future proofing.

Reply:
Yeah it's a managed prefix list by EC2 and you cannot even delete it on your own. Ideally there should be only one for cloudfront per region and they always have the same id same name for every account (same region).

Reply:
Gotcha. My concern was just that, if the expectation of "exactly one prefix list" is unique to CloudFront, perhaps we should handle it in the caller that calls `ManagedPrefixListID(cloudFrontPrefixListName)` instead of here. No biggie.  Happy to approve - if it happen in the future that a caller to this method expect it to faithfully return all prefix list, we can do a refactor which is not going to be difficult.

Reply:
Sounds good to me